lib/internal/encoding.js

```diff
@@ -344,8 +355,7 @@ function makeTextDecoderICU() {
   class TextDecoder {
     constructor(encoding = 'utf-8', options = {}) {
       encoding = `${encoding}`;
-      if (typeof options !== 'object')
-        throw new errors.Error('ERR_INVALID_ARG_TYPE', 'options', 'Object');
+      validateArgument(options, 'object', 'options');
```

Comment:
By combining the check for the type with the second `ERR_INVALID_ARG_TYPE` argument the error message gets changed. That should not be the case. That applies to all `validateArgument` calls.

Reply:
@BridgeAR the `string` type checks error messages is not affected.